### PR TITLE
fix: warn about missing descriptions and return "" not `None`

### PR DESCRIPTION
**Pull request by JoeHCQ1**
This should fix https://github.com/mkdocstrings/mkdocstrings/pull/397 and #137 .

The problem was that empty descriptions were being returned as None in `Parameters` and `Exceptions` blocks instead of `""`. 

That None eventually made it to this: <https://github.com/mkdocstrings/mkdocstrings/blob/master/src/mkdocstrings/handlers/base.py#L219>

And that passed it here: <https://github.com/Python-Markdown/markdown/blob/master/markdown/core.py#L225>

And later in that function we get a Nonetype error: <https://github.com/Python-Markdown/markdown/blob/master/markdown/core.py#L248>

Because we also were not recording empty descriptions as errors (which I believe mkdocstrings interprets and reports as warnings) there was nothing to tell the user _why_  `python -m mkdocs build` had thrown a Nonetype error.

So I:
- Recorded missing descriptions as errors.
- Changed the code to return all missing descriptions as `""` instead of `None`.
- Added test to verify capability.

### CODE REVIEW

File: src/pytkdocs/parsers/docstrings/numpy.py [hunk 1]
```diff
@@ -89,11 +89,16 @@ def read_parameters_section(
                 if signature_param.default is not empty:
                     default = signature_param.default
                 kind = signature_param.kind
+            
+            description = param.description if param.description else ""
```
Comment (pawamoy) [inline src/pytkdocs/parsers/docstrings/numpy.py:93]:
```suggestion
            description = param.description or ""
```

File: src/pytkdocs/parsers/docstrings/numpy.py [hunk 2]
```diff
@@ -124,6 +129,9 @@ def read_attributes_section(
         docstring_attributes = [p for p in docstring_obj.params if p.args[0] == "attribute"]
 
         for attr in docstring_attributes:
+            description = attr.description if attr.description else ""
```
Comment (pawamoy) [inline src/pytkdocs/parsers/docstrings/numpy.py:132]:
```suggestion
            description = attr.description or ""
```

File: src/pytkdocs/parsers/docstrings/numpy.py [hunk 3]
```diff
@@ -157,6 +165,9 @@ def read_exceptions_section(
         except_obj = docstring_obj.raises
 
         for exception in except_obj:
+            description = exception.description if exception.description else ""
```
Comment (pawamoy) [inline src/pytkdocs/parsers/docstrings/numpy.py:168]:
```suggestion
            description = exception.description or ""
```

File: tests/test_parsers/test_docstrings/test_numpy.py [hunk 4]
```diff
@@ -81,6 +81,37 @@ def test_sections_without_signature():
...
+    sections, errors = parse(
+        """
+        Sections without descriptions.
+
+        Parameters
+        ----------
+        void : str
+        niet : str
+        nada : str
+        rien : str
```
Comment (pawamoy) [inline tests/test_parsers/test_docstrings/test_numpy.py:96]: You can probably use only one parameter, that would be enough.

File: tests/test_parsers/test_docstrings/test_numpy.py [hunk 5]
```diff
@@ -81,6 +81,37 @@ def test_sections_without_signature():
...
+        -------
+        bool
+        """
+    )
+    assert len(sections) == 4
+    assert len(errors) == 10
+    for error in errors[:8]:
+        assert "param" in error
+    assert "exception" in errors[8]
+    assert "return description" in errors[9]
```
Comment (pawamoy) [inline tests/test_parsers/test_docstrings/test_numpy.py:112]: Could you also test for each section item that its value is indeed an empty string and not none?

**Review by pawamoy — COMMENTED** [review]
Perfect investigation! 

Thanks a lot for taking the time to find and fix the root cause of the issue :slightly_smiling_face: 

I've made a few suggestions and comments.

File: src/pytkdocs/parsers/docstrings/numpy.py (hunk 1 above)
Comment (JoeHCQ1) [reply src/pytkdocs/parsers/docstrings/numpy.py]: Ah, I thought there should be a easier way to say that. Thanks for the the tip.

File: tests/test_parsers/test_docstrings/test_numpy.py (hunk 4 above)
Comment (JoeHCQ1) [reply tests/test_parsers/test_docstrings/test_numpy.py:96]: One would probably be ok, four is definitely more than necessary. I'm changing this to two parameters because I want to make sure it works with one as well as multiple. The Exceptions and Returns sections verify the one parameter case. Knowing the implementation, I don't know that this is necessary, but I like to ignore my knowledge of the implementation to some extent when writing tests.

**Comment by JoeHCQ1** [comment]
@pawamoy I am ready for re-review.

File: tests/test_parsers/test_docstrings/test_numpy.py (hunk 4 above)
Comment (pawamoy) [reply tests/test_parsers/test_docstrings/test_numpy.py]: Very valid point! Thanks for expressing it :slightly_smiling_face:

**Review by pawamoy — APPROVED** [review]
LGTM!

**Comment by pawamoy** [comment]
Thanks a lot! Will release it now.

**Comment by pawamoy** [comment]
Released in 0.16.1
